ansible/configs/multi-region-tower/files/cloud_providers/worker.j2

```diff
@@ -0,0 +1,369 @@
+
+  VpcGA:
+    Type: "AWS::EC2::VPCGatewayAttachment"
+    Properties:
+      InternetGatewayId:
+        Ref: VpcInternetGateway
+      VpcId:
+        Ref: Vpc
+
+  PublicSubnet:
+    Type: "AWS::EC2::Subnet"
+    DependsOn:
+      - Vpc
+    Properties:
+    {% if aws_availability_zone is defined %}
+      AvailabilityZone: {{ aws_availability_zone }}
+    {% endif %}
+
+      CidrBlock: "{{ aws_public_subnet_cidr }}"
+      Tags:
+        - Key: Name
+          Value: "{{project_tag}}"
+        - Key: Hostlication
+          Value:
+            Ref: "AWS::StackId"
+      MapPublicIpOnLaunch: true
+      VpcId:
+        Ref: Vpc
+
+  PublicSubnetRTA:
+    Type: "AWS::EC2::SubnetRouteTableAssociation"
+    Properties:
+      RouteTableId:
+        Ref: VpcRouteTable
+      SubnetId:
+        Ref: PublicSubnet
+
+{% for security_group in security_groups|list + default_security_groups|list %}
+  {{security_group['name']}}:
+    Type: "AWS::EC2::SecurityGroup"
+    Properties:
+      GroupDescription: Host
+      VpcId:
+        Ref: Vpc
+      Tags:
+        - Key: Name
+          Value: "{{security_group['name']}}"
+{% endfor %}
+
+{% for security_group in default_security_groups|list + security_groups|list %}
+{% for rule in security_group.rules %}
+  {{security_group['name']}}{{rule['name']}}:
+    Type: "AWS::EC2::SecurityGroup{{rule['rule_type']}}"
+    Properties:
+     GroupId:
+       Fn::GetAtt:
+         - "{{security_group['name']}}"
+         - GroupId
+     IpProtocol: {{rule['protocol']}}
+     FromPort: {{rule['from_port']}}
+     ToPort: {{rule['to_port']}}
+  {% if rule['cidr'] is defined %}
+     CidrIp: "{{rule['cidr']}}"
+  {% endif  %}
+
+  {% if rule['from_group'] is defined %}
+     SourceSecurityGroupId:
+       Fn::GetAtt:
+        - "{{rule['from_group']}}"
+        - GroupId
+  {% endif  %}
+{% endfor %}
+{% endfor %}
+
+  DnsZonePrivate:
+    Type: "AWS::Route53::HostedZone"
+    Properties:
+      Name: "{{ aws_dns_zone_private }}"
+      VPCs:
+        - VPCId:
+            Ref: Vpc
+          VPCRegion:
+            Ref: "AWS::Region"
+      HostedZoneConfig:
+        Comment: "{{ aws_comment }}"
+
+  {% if secondary_stack is not defined %}
+  DnsZonePublic:
+    Type: "AWS::Route53::HostedZone"
+    Properties:
+      Name: "{{ aws_dns_zone_public }}"
+      HostedZoneConfig:
+        Comment: "{{ aws_comment }}"
+
+  DnsPublicDelegation:
+    Type: "AWS::Route53::RecordSetGroup"
+    DependsOn:
+      - DnsZonePublic
+    Properties:
+    {% if HostedZoneId is defined %}
+      HostedZoneId: "{{ HostedZoneId }}"
+    {% else %}
+      HostedZoneName: "{{ aws_dns_zone_root }}"
+    {% endif %}
+      RecordSets:
+        - Name: "{{ aws_dns_zone_public }}"
+          Type: NS
+          TTL: {{ aws_dns_ttl_public }}
+          ResourceRecords:
+            "Fn::GetAtt":
+              - DnsZonePublic
+              - NameServers
+    {% endif %}
+
+{% for instance in instances_worker %}
```

Comment:
That is the **only** difference between `default.j2` and your other template `worker.j2`

```diff
 
-{% for instance in instances %}
+{% for instance in instances_worker %}
 {% if instance['dns_loadbalancer'] | d(false) | bool
   and not instance['unique'] | d(false) | bool %}
```

We should not need 2 nearly identical templates just for that.

The capability of specifying the region per instance in `instances` i wrote about earlier would solve this for you.

OR

You can override the `instances` var when calling the  `template` ansible module
```yaml
        stack_instances: "{{ instances if 'instances_var' not in _region else vars[_region.instances_var] }}"
```

in `post_infra.yml`:


```yaml
    - name: Deploy secondary stacks
      loop: "{{ target_regions[1:] }}"
      vars:
        stack_file: "{{ _region.stack }}"
        stack_instances: "{{ instances if 'instances_var' not in _region else vars[_region.instances_var] }}"
        aws_region: "{{ _region.region }}"
        aws_vpc_cidr: "{{ _region.vpc_cidr }}"
        aws_public_subnet_cidr: "{{ _region.subnet_cidr }}"
        aws_dns_zone_private: "{{ _region.name }}.{{ guid }}.internal."
        aws_dns_zone_public_prefix: "{{ _region.name }}."
        secondary_stack: yes
      loop_control:
        loop_var: _region
      include_tasks: deploy_stack.yml
```
Then in `deploy_stack.yml`

```yaml
    - name: Run infra-ec2-template-generate Role (target_regions)
      include_role:
        name: infra-ec2-template-generate
      vars:
        cloudformation_template: "{{ output_dir }}/{{ env_type }}.{{ guid }}.{{ cloud_provider }}_cloud_template_{{ aws_region }}"
        instances: "{{ stack_instances }}"
```

(but just adding the filter is easier in my opinion)